### DISCUSSION TITLE
Flat rate tax plugin

### DISCUSSION THREAD

**Discussion by gsvr** [saleor/saleor#7942] [Ideas]
Hi! I live in a country (ZA) where taxes on products is really simple - you either pay 15% or 0%.

I'm planning on writing a tax plugin to support this for our shop in the next month or two, but thought that it might be something that's applicable to other countries as well.

If it is, I thought it might be worth me attempting to write a more generic plugin that can be added to saleor core for anyone to use, rather than just my own plugin for our use. But I reckon it could also be an external plugin that someone could grab from my github page.

Any advice on what the most helpful approach would be here?

Also, maybe it's just my country that does things this way and it wouldn't help anyone else much if I added something like this, if you have any knowledge around that please let me know :)

**Comment by patrys** [comment]
This is something we want to address. So our internal plan was to move the tax calculation logic from the VatLayer plugin into Saleor itself and only make that plugin responsible for synchronizing the individual tax rates. However, we are not actively working on this, so if it looks like something you'd be willing to contribute, we can help you refine this change to the point where it gets accepted into the core.

**Reply by gsvr** [reply]
I'd be keen to contribute, I'll check with our CTO about doing it on company time. If that's not high priority I'll see if I can work on it in my spare time, but it could take a fair bit longer due to limited free time. I'll probably have more clarity in the next couple of weeks and will respond back here :)

**Reply by tmechsner** [reply]
Any news on this? I'd love this extension too, as we have lots of issues with the VatLayer integration and only need the two tax rates of Germany.

**Reply by Juleszz** [reply]
I'd suggest to handle taxes outside of Saleor. I was having a lot of problems with VatLayer, too - the tax laws are simply a lot more complicated than the solution VatLayer is offering. Also an interal logic might cause problems depending on the country you live in and what type of clients you have. If you really just need a simple tax setup for one country I'd develop it myself.

E.g. in our case we show taxes to the customer that are calculated by the frontend. Saleor is accepting orders and payments with gross amounts and all data will be evaluated in our internal backend (in this case NINOX). This is where invoices etc. are generated and tax rules apply.

It's actually pretty simple and save and this even works with B2C and B2B customers while working on different channels.

**Comment by carlosa54** [comment]
Hi @gsvr, I think this https://github.com/saleor/saleor/discussions/8341 might be able to cover your use case.
